test/src/mocksocket.ts

```diff
@@ -82,21 +77,21 @@ class SocketBase {
    */
   triggerClose() {
     this._readyState = SocketBase.CLOSING;
-    if (this._onClose) {
-      utils.doLater(() => {
-        this._readyState = SocketBase.CLOSED;
-        this._onClose();
-      });
-    }
+    utils.doLater(() => {
+      this._readyState = SocketBase.CLOSED;
+      var onClose = this._onClose;
+      if (onClose) onClose();
```

Comment:
This one has different logic than before, changing the `_readyState` whether or not `this._onClose` is defined.